manual/manual/html_processing/js/search.js

```diff
@@ -0,0 +1,242 @@
+	err = MAX_ERROR;
+    } else {
+	if ( sub.includes("->") ) {
+	    err = Math.min(s.length - sub.length - err,1); // 0 or 1
+	    // err = 0 if the substring is right-aligned
+	} else {
+	    err = Math.min(err,1); // 0 or 1
+	    // err = 0 if the substring
+	}
+	err += Math.abs((s.length - sub.length) / s.length);}
+    return (err)
+    // between 0 and 2, except if MAX_ERROR
+}
+
+// Minimal substring error. In particular, it returns 0 if the string
+// 'sub' has an exact match with one of the strings in 'line'.
+function subMinError (sub, line) {
+    //let strings = [line[0], line[1], line[2]];
+    let errs = line.map(function (s) { return subError (sub, s); });
+    return Math.min(...errs); // destructuring assignment
+}
+
+
+function add (acc, a) {
+    return acc + a;
+}
+
+// for each sub we compute the minimal error within 'line', and then
+// take the average over all 'subs'. Thus it returns 0 if each sub has
+// an exact match with one of the strings in 'line'.
+function subsMinError (subs, line) {
+    let errs = subs.map(function (sub) { return subMinError (sub, line); });
+    return errs.reduce(add,0) / subs.length;
+}
+    
+
+function formatLine (line) {
+    let li = '<li>';
+    let html = '<code class="code"><a href="' + line[1] + '"><span class="constructor">' + line[0] + '</span></a>' +
+	'.' + '<a href="' + line[3] + '">' + line[2] + '</a></code>';
+    if (line.length > 5) {
+	if ( line[ERR_INDEX] == 0 ) {
+	    li = '<li class="match">';
+	}
+	html += (' : ' + line[SIG_INDEX] + '</pre>');
+	html = '<pre>' + html + line[DESCR_INDEX]; }
+    return (li + html + "</li>\n");
+}
+
+// The initial format of an entry of the GENERAL_INDEX array is
+// [ module, module_link,
+//   value, value_link,
+//   html_description, bare_description,
+//   html_signature, bare_signature ]
+
+// If includeDescr is false, the line is truncated to its first 4
+// elements.  When searching, the search error is added at the end of
+// each line.
+
+// In order to reduce the size of the index.js file, one could create
+// the bare_description on-the-fly using .textContent, see
+// https://stackoverflow.com/questions/28899298/extract-the-text-out-of-html-string-using-javascript,
+// but it would probably make searching slower (haven't tested).
+function mySearch (includeDescr) {
+    if (loadingIndex (includeDescr)) {
+	return;
+    }
+    let text = document.getElementById('api_search').value;
+    let results = [];
+    let html = "";
+    let count = 0;
+    let err_index = DESCR_INDEX;
+
+    if (text !== "") {
+	let words = [];
+	if ( includeDescr ) {
+	    err_index = ERR_INDEX;
+	    
+	    // Split text into an array of non-empty words:
+	    if ( text.includes("->") ) {
+		// this must be a type, so we don't split on single space
+		words = text.split("  ");
+	    } else {
+		words = text.split(" ");
+	    };
+	    words = words.filter(function (s) {
+		return (s !== "")});
+	    console.log (words);
```

Comment:
The console logging should probably disappear.

Reply:
This was already removed, you're probably looking at an older version

Reply:
but there is another console.log `console.log("Search results = " + (count.toString()));`
which can be removed too if you want; although this one is only one integer, so not too invasive.